packages/html-reporter/src/chip.tsx

```diff
@@ -53,17 +53,24 @@ export const AutoChip: React.FC<{
   noInsets?: boolean,
   children?: any,
   dataTestId?: string,
-  targetRef?: React.RefObject<HTMLDivElement>,
-}> = ({ header, initialExpanded, noInsets, children, dataTestId, targetRef }) => {
-  const [expanded, setExpanded] = React.useState(initialExpanded || initialExpanded === undefined);
-  return <Chip
-    header={header}
-    expanded={expanded}
-    setExpanded={setExpanded}
-    noInsets={noInsets}
-    dataTestId={dataTestId}
-    targetRef={targetRef}
-  >
-    {children}
-  </Chip>;
+  anchorId?: string,
+}> = ({ header, initialExpanded, noInsets, children, dataTestId, anchorId }) => {
+  const [expanded, setExpanded] = React.useState(initialExpanded ?? true);
+  const onChangeReveal = React.useCallback((isRevealed: boolean) => {
+    if (isRevealed)
+      setExpanded(true);
```

Comment:
Why do we need this logic? If it is collapsed by default, perhaps it's fine for it to be collapsed after scrolling into view?

Reply:
UX-Wise, I think uncollapsing on reveal saves one click. This might not make a difference for Traces or Videos. But we'll extend this to a specific attachment in the future, and there's no way of revealing an element in a collapsed Chip.

Reply:
Thinking about this one, it means in addition to scrolling into view, we might need to perform other actions to reveal, for example expand the chip.

If we think broadly, the way to solve it would be to add revealing logic to various UI elements:
- Make expandable/switchable elements accept `anchorId` that they will `useAnchor()` with and reveal the respective part of the UI.
- Separate `Anchor` that will scroll into view from the `AutoChip` that will expand, and pass the same `anchorId` to both.
- Make `anchorId` hierarchical in the future, so that we can in theory say "AutoChip that contains all attachments has anchorId=attachments/*", and then inside it put individual "Attachment with anchorId=attachments/0".

Let me know what you think.

Reply:
You've put into words what I couldn't, yes! That's why `useAnchor` is probably gonna be exported in the next one, and why `anchorId` accepts a predicate. I'll look at separating the expanding + scrolling logic in AutoChip.